File: DAM 3er_año_1º-2025-26/Programación/UD04 - Ej1/UD04 - EJ1/src/Ej2.java
/*
2. Crear una función que calcule el menor de dos números enteros que recibirá
como parámetros. El resultado será otro número entero.
*/

import java.util.Scanner;

public class Ej2 {
    public static int menorDosNumeros(int n1, int n2) {
        return Math.min(n1, n2);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Dime un número: ");
        int n = scanner.nextInt();
        scanner.nextLine();

        System.out.print("Dime otro numero: ");
        int n1 = scanner.nextInt();
        scanner.nextLine();

        System.out.println("El numero menor es: " + menorDosNumeros(n, n1));

    }
}

File: DAM_2º-año_1º/Programación/UD7 - Kotlin Ej1/src/Ej2.kt
/**
 * 2. Calcula el factorial de un número usando una función recursiva.
 * **/
fun recursiva(n: Int):Long{
    return if (n<= 1) 1
    else n*recursiva(n-1)
}
fun main() {
    print("Dime un número: ")
    val n = readln().toInt();

    print("El factorial de $n es ${recursiva(n)}")
}